### ISSUE TITLE
Flaky test: TestManagerConfigPeriodicRefresh

### ISSUE THREAD

**Issue by rturner3** [spiffe/spire#3401]
```
2022/09/01 19:33:15 http: TLS handshake error from 127.0.0.1:45612: read tcp 127.0.0.1:33971->127.0.0.1:45612: use of closed network connection
--- FAIL: TestManagerConfigPeriodicRefresh (0.02s)
    manager_test.go:184: 
        	Error Trace:	/home/runner/work/spire/spire/pkg/server/bundle/client/manager_test.go:184
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 2
        	Test:       	TestManagerConfigPeriodicRefresh
FAIL
FAIL	github.com/spiffe/spire/pkg/server/bundle/client	0.184s
```

[Build link](https://github.com/spiffe/spire/runs/8143312825?check_suite_focus=true)

**Comment by xiaods** [comment]
@rturner3  please double check upstream code. in my env ,it's passed.
```
Running tool: /usr/local/bin/go test -timeout 30s -run ^TestManagerConfigPeriodicRefresh$ github.com/spiffe/spire/pkg/server/bundle/client -v

=== RUN   TestManagerConfigPeriodicRefresh
--- PASS: TestManagerConfigPeriodicRefresh (0.01s)
PASS
ok  	github.com/spiffe/spire/pkg/server/bundle/client	0.533s
```

**Comment by azdagron** [comment]
Failure in this case is dependent on timing. If you run the test enough, it will fail. I usually do something like this:
```
$ cd pkg/server/bundle/client
$ go test -c
$ while ./client.test; do :; done
```
And then wait. Eventually this will trigger a failure.

The reason these tests fail (there is more than one) is because when a trust domain becomes configured, a goroutine is started that will update the bundle for that trust domain. Some of the tests don't handle that correctly and assume there will just be a single update, but depending on when that goroutine gets scheduled, there might be another. It's all based on the scheduler and timing.

**Comment by xiaods** [comment]
> Failure in this case is dependent on timing. If you run the test enough, it will fail. I usually do something like this:
> 
> ```
> $ cd pkg/server/bundle/client
> $ go test -c
> $ while ./client.test; do :; done
> ```
> 
> And then wait. Eventually this will trigger a failure.
> 
> The reason these tests fail (there is more than one) is because when a trust domain becomes configured, a goroutine is started that will update the bundle for that trust domain. Some of the tests don't handle that correctly and assume there will just be a single update, but depending on when that goroutine gets scheduled, there might be another. It's all based on the scheduler and timing.

when i run client.test, it only output below log:
```
PASS
2022/09/14 07:23:56 http: TLS handshake error from 127.0.0.1:57269: remote error: tls: bad certificate
2022/09/14 07:23:56 http: TLS handshake error from 127.0.0.1:57275: remote error: tls: bad certificate
...strips...
```

**Comment by azdagron** [comment]
How many times did you run it?

**Comment by xiaods** [comment]
> How many times did you run it?
which step need run many times? i have run  below the step many times. 
```
$ while ./client.test; do :; done
```
I have not seen your report errors.

**Comment by azdagron** [comment]
You've won the timing lottery :) I can reproduce this within a minute or two on my machine. It's also reproducing in the CI/CD pipeline.

**Comment by azdagron** [comment]
Fixed by #3575
